herddb-jdbc/src/test/java/herddb/jdbc/SystemTablesTest.java

```diff
@@ -227,4 +225,148 @@ public void test() throws Exception {
             }
         }
     }
+
+    @Test(expected = SQLException.class)
+    public void ensureWeThrowExceptionForNonSupportedNotNullTypes_Double() throws Exception {
+        try (Server server = new Server(new ServerConfiguration(folder.newFolder().toPath()))) {
+            server.start();
+            server.waitForStandaloneBoot();
+            try (HDBClient client = new HDBClient(new ClientConfiguration(folder.newFolder().toPath()));) {
+                client.setClientSideMetadataProvider(new StaticClientSideMetadataProvider(server));
+                try (BasicHerdDBDataSource dataSource = new BasicHerdDBDataSource(client);
+                     Connection con = dataSource.getConnection();
+                     Statement statement = con.createStatement();) {
+                     statement.execute("CREATE TABLE mytable (key string primary key, name string,d1 double not null)");
+                }
+            }
+        }
+    }
+
+    @Test(expected = SQLException.class)
+    public void ensureWeThrowExceptionForNonSupportedNotNullTypes_Float() throws Exception {
+        try (Server server = new Server(new ServerConfiguration(folder.newFolder().toPath()))) {
+            server.start();
+            server.waitForStandaloneBoot();
+            try (HDBClient client = new HDBClient(new ClientConfiguration(folder.newFolder().toPath()));) {
+                client.setClientSideMetadataProvider(new StaticClientSideMetadataProvider(server));
+                try (BasicHerdDBDataSource dataSource = new BasicHerdDBDataSource(client);
+                     Connection con = dataSource.getConnection();
+                     Statement statement = con.createStatement();) {
+                    statement.execute("CREATE TABLE mytable (key string primary key, name string,f1 float not null)");
+                }
+            }
+        }
+    }
+
+    @Test(expected = SQLException.class)
+    public void ensureWeThrowExceptionForNonSupportedNotNullTypes_BOOLEAN() throws Exception {
```

Comment:
we can keep just this one and a test which "works", there is no much value in testing other datatype